src/k2/httpproxy/HTTPProxy.cpp

```diff
@@ -378,6 +378,31 @@ seastar::future<nlohmann::json> HTTPProxy::_handleWrite(nlohmann::json&& request
     });
 }
 
+seastar::future<std::tuple<k2::Status,dto::CreateSchemaResponse>> HTTPProxy::_handleCreateSchema(
+    dto::CreateSchemaRequest&& request) {
+    K2LOG_D(k2::log::httpproxy, "Received create schema request {}", request);
+    return _client.createSchema(std::move(request.collectionName), std::move(request.schema))
+        .then([] (CreateSchemaResult&& result) {
+            return seastar::make_ready_future<std::tuple<k2::Status,dto::CreateSchemaResponse>>(
```

Comment:
This is where we use the helper func from the dispatcher. Instead of the seastar::make_future....., just use:
```
return RPCResponse(std::move(result.status), dto::CreateSchemaResponse{});
```

Reply:
Done. It makes it much cleaner!